components/sup/src/manager/service.rs

```diff
@@ -810,25 +814,36 @@ impl Service {
         }
     }
 
-    fn post_run(&mut self) {
-        let _timer = hook_timer("post-run");
-
+    fn post_run(&mut self, executor: &TaskExecutor) {
         if let Some(ref hook) = self.hooks.post_run {
-            hook.run(&self.service_group,
-                     &self.pkg,
-                     self.svc_encrypted_password.as_ref());
+            let hook_runner = HookRunner::new(Arc::clone(&hook),
+                                              self.service_group.clone(),
+                                              self.pkg.clone(),
+                                              self.svc_encrypted_password.clone());
+            let (handle, f) = hook_runner.loop_future();
+            self.post_run_handle = Some(handle);
+            executor.spawn(f);
+        }
+    }
+
+    /// Stop the `post-run` future. The `post-run` hook has retry logic based on its exit code. This
+    /// will stop this retry loop regardless of `post-run`'s exit code.
+    ///
+    /// Consumes the handle to that future in the process.
```

Comment:
Is there a way to indicate that in the name of the method rather than the comment? Or maybe a more interesting question is: does the caller of the method need to know this?